internal/tools/aws/database_multitenant.go

```diff
@@ -973,41 +971,40 @@ func (d *RDSMultitenantDatabase) cleanupDatabase(rdsClusterID, rdsClusterendpoin
 func (d *RDSMultitenantDatabase) ensureMultitenantDatabaseSecretIsCreated(rdsClusterID, VpcID *string) (*RDSSecret, error) {
 	installationSecretName := RDSMultitenantSecretName(d.installationID)
 
-	installationSecretValue, err := d.client.Service().secretsManager.GetSecretValue(&secretsmanager.GetSecretValueInput{
-		SecretId: aws.String(installationSecretName),
-	})
-	if err != nil && !IsErrorCode(err, secretsmanager.ErrCodeResourceNotFoundException) {
+	installationSecret, err := d.client.secretsManagerGetRDSSecret(installationSecretName, d.client.logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get secret value for database")
+	}
+
+	var awsErr *smTypes.ResourceNotFoundException
+	if err != nil && !errors.As(err, &awsErr) {
```

Comment:
Can you revise this logic? Looks like unreachable.

Reply:
Thanks for your 👀 Mir, please check if [this has more sense](https://github.com/mattermost/mattermost-cloud/pull/802/commits/fdf329494795cbeb066c2c1199bb39e2af1eba82), I think I got it right now.